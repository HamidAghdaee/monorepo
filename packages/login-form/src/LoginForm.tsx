import { Input } from '@taxi/input';
import * as React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 10px;
`;

const ButtonsWrapper = styled.div`
  text-align: right;
`;

export interface LoginFormProps {
  onClick?: () => void;
}

export const LoginForm: React.SFC<LoginFormProps> = ({ onClick }) => (
  <Wrapper>
    <Input id="name" label="Name" />
    <Input id="password" label="Password" />

    <ButtonsWrapper>
      <button onClick={onClick}>Log in</button>
    </ButtonsWrapper>
  </Wrapper>
);
