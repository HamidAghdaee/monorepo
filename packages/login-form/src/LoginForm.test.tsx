import { mount } from 'enzyme';
import * as React from 'react';

import { LoginForm } from './LoginForm';

jest.mock('@taxi/input', () => ({ Input: 'input' }));

describe('LoginForm', () => {
  test('should match snapshot and styles', () => {
    expect(mount(<LoginForm />)).toMatchSnapshot();
  });
});
