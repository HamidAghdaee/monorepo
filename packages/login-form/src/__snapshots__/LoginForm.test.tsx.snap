// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot and styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10px;
}

.c1 {
  text-align: right;
}

<Component>
  <styled.div>
    <div
      className="c0"
    >
      <input
        id="name"
        label="Name"
      />
      <input
        id="password"
        label="Password"
      />
      <styled.div>
        <div
          className="c1"
        >
          <button>
            Log in
          </button>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Component>
`;
